concepts/0020-message-types/README.md

```diff
@@ -0,0 +1,223 @@
+# 0020: Message Types
+- Name: message-types
+- Authors: Daniel Bluhm <daniel.bluhm@sovrin.org>, Sam Curren <sam@sovrin.org>
+- Start Date: 2018-07-06
+
+## Status
+
+- Status: [ACCEPTED](/README.md#rfc-lifecycle)
+- Status Date: 2019-05-24
+- Status Note: Supersedes [HIPE 0021 Message Types](https://github.com/hyperledger/indy-hipe/tree/master/text/0021-message-types)
+
+## Summary
+
+Define structure of message type strings used in agent to agent communication, describe their resolution to documentation URIs, and offer guidelines for type family specifications.
+
+## Motivation
+
+A clear convention to follow for agent developers is necessary for interoperability and continued progress as a community.
+
+## Tutorial
+
+A "Message Type" is a required attribute of all communications sent between parties. The message type instructs the receiving agent how to interpret the content and what content to expect as part of a given message.
+
+Types are specified within a message using the `@type` attribute:
+
+```json
+{
+    `@type`:<message type string>,
+    // other attributes
+}
+```
+
+Message types are URIs that resolve to developer documentation for the message type. The end portion of the URI MUST follow the following format:
+
+```
+/<message_family>/<major_family_version>.<minor_family_version>/<message_type>
```

Comment:
I would be happier describing the first section as protocol, not message family. Also, note that this content is redundant with a discussion about versioning in https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/uris.md and https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/semver.md. We should probably link to those docs instead of repeating the content here.

Reply:
Did I sufficiently fix this? I left the example in play. Also, message families -> protocols.